tensorvis/tensorvisio/src/processors/vtktotensorfield2d.cpp

```diff
@@ -0,0 +1,128 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/tensorvisio/processors/vtktotensorfield2d.h>
+
+#include <vtkPointData.h>
+#include <vtkDataArray.h>
+#include <vtkDoubleArray.h>
+#include <vtkFloatArray.h>
+
+#include <modules/tensorvisbase/datastructures/tensorfield2d.h>
+#include <modules/tensorvisbase/util/misc.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo VTKToTensorField2D::processorInfo_{
+    "org.inviwo.VTKToTensorField2D",  // Class identifier
+    "VTK To Tensor Field 2D",         // Display name
+    "VTK",                            // Category
+    CodeState::Experimental,          // Code state
+    "VTK",                            // Tags
+};
+const ProcessorInfo VTKToTensorField2D::getProcessorInfo() const { return processorInfo_; }
+
+VTKToTensorField2D::VTKToTensorField2D()
+    : Processor()
+    , vtkDataSetInport_("vtkDataSetInport")
+    , tensorField2DOutport_("tensorField2DOutport") {
+    addPort(vtkDataSetInport_);
+    addPort(tensorField2DOutport_);
+}
+
+void VTKToTensorField2D::process() {
+    const auto inviwoDataSet = vtkDataSetInport_.getData();
+    const auto& vtkDataSet = *vtkDataSetInport_.getData();
+
+    size3_t dimensions{};
+    if (auto d = inviwoDataSet->getDimensions())
+        dimensions = *d;
+    else {
+        LogError("VTK data set type not supported.");
+        return;
+    }
+    if (dimensions.z != 1) {
+        LogError("Z dimension != 1. Aborting.");
+        return;
+    }
+
+    const auto numberOfElements = glm::compMul(dimensions);
+
+    const auto pointData = vtkDataSet->GetPointData();
+    const auto tensors = pointData->GetTensors();
+
+    if (tensors->GetDataType() != VTK_FLOAT && tensors->GetDataType() != VTK_DOUBLE) {
+        LogError("Data type is neither float nor double. Aborting.");
```

Comment:
exception

Reply:
what is the reason for sticking to float/double? Is it possible to dispatch from vtk side?

Reply:
Mainly two reasons:
1. The tensor field class is not templated (always stores doubles)
2. I would argue that in the case of tensors you will never encounter anything other than floats or doules.